form.go

```diff
@@ -129,13 +130,26 @@ type Field interface {
 	// WithHeight sets the height of a field.
 	WithHeight(int) Field
 
+	// WithPosition tells the field the index of the group and position it is in.
+	WithPosition(Position) Field
+
 	// GetKey returns the field's key.
 	GetKey() string
 
 	// GetValue returns the field's value.
 	GetValue() any
 }
 
+// Position is positional information about the given field and form.
+type Position struct {
```

Comment:
since all fields are private, does this struct, and `WithPosition`, need to be public? 🤔

Reply:
Exposed all the fields since this is an interface and could be used by others I think we need to make these public.